### ISSUE TITLE
Dirty attributes dont get updated

### ISSUE THREAD

**Issue by m4c3** [flexirest/flexirest#131]
### Expected behaviour

i update a nested result iterator, the attributes get marked as unsaved but they wont be saved

### Desired behaviour

dirty attributes of a nested result Iterator should get updated

### How to reproduce

1. Clone the Flexirest repo (`git clone https://github.com/flexirest/flexirest.git`)
2. Change to the flexirest folder (`cd flexirest`)
3. Install any dependencies (`bundle install`)
4. Run an interactive console (`rake c`)
5. Paste the following:

```ruby
class MyObject < Flexirest::Base
  verbose true
  base_url "https://dontcare.free.beeceptor.com"
  get :get, "/123"
  post :update, "/123", only_changed: true
end

my_object = MyObject.new()
my_object.get
puts my_object.inspect
puts my_object.userdata.devicetype
my_object.userdata.devicetype="something"
puts my_object.userdata.devicetype
puts my_object.inspect
puts my_object.changed?
puts my_object.changes
my_object.update
```

**Comment by andyjeffries** [comment]
Thanks for this @m4c3 , this is a really good bug report. And actually a tricky one to fix I think. I'll have a play.

**Comment by andyjeffries** [comment]
I don't know if it works with your `update` call (I haven't really seen anyone using Flexirest for updated nested resources), but at least the `changed?` and `changes` report that it's changed now (not the nested actually change but at least the attribute has changed and IT reports its own changes). Can you try using the master branch and see if it fixes it for your use case please.

**Comment by m4c3** [comment]
Thanks! This works now 👍

**Comment by andyjeffries** [comment]
Thanks, I've released this as v0.10.1 now then.
